.github/workflows/checks-and-builds.yaml

```diff
@@ -20,34 +20,19 @@ jobs:
           fetch-depth: 0
       - name: Check style
         run: "./ci/check_style.sh"
-  conda-build:
+  build-conda:
     needs: check-style
-    runs-on: ubuntu-latest
-    container:
-      image: rapidsai/ci-conda:cuda12.2.2-ubuntu22.04-py3.9
-    env:
-      PUBLISH: "${{ inputs.publish }}"
-    steps:
-      - uses: actions/checkout@v4
-      - name: Set Proper Conda Upload Token
-        run: |
-          RAPIDS_CONDA_TOKEN=${{ secrets.CONDA_RAPIDSAI_NIGHTLY_TOKEN }}
-          if rapids-is-release-build; then
-            RAPIDS_CONDA_TOKEN=${{ secrets.CONDA_RAPIDSAI_TOKEN }}
-          fi
-          echo "RAPIDS_CONDA_TOKEN=${RAPIDS_CONDA_TOKEN}" >> "${GITHUB_ENV}"
-      - name: Python build
-        run: "./ci/build_python.sh ${PUBLISH}"
-  wheel-build:
+    uses: rapidsai/shared-workflows/.github/workflows/conda-python-build.yaml@branch-24.06
+    with:
+      build_type: branch
+      script: "ci/build_python.sh"
+      # Select only the build with the minimum Python version and the maximum CUDA version
+      matrix_filter: map(select(.ARCH == "amd64")) | group_by(.CUDA_VER|split(".")|map(tonumber)|.[0]) | map(min_by([(.PY_VER|split(".")|map(tonumber)), (.CUDA_VER|split(".")|map(tonumber))])) | [max_by(.PY_VER|split(".")|map(tonumber))]
```

Comment:
I'm not a jq expert. Is there a simpler way to do the last two steps? I wonder if you can do "min_by (Python version, then _negative_ CUDA version)" so that you get the desired output in one step.

ChatGPT proposes the following. Maybe replace `sort_by(...)[0]` with `min_by`.

```
map(select(.ARCH == "amd64")) 
| sort_by(
    (.PY_VER | split(".") | map(tonumber)),      # Ascending sort by Python version
    -(.CUDA_VER | split(".") | map(tonumber))    # Descending sort by negating the CUDA version
  )[0]
```

Reply:
Very close. This ended up being the winner:

```
[
  map(select(.ARCH == "amd64"))
  | min_by(
    (.CUDA_VER | split(".") | map(tonumber) | map(-.)),
    (.PY_VER | split(".") | map(tonumber))
  )
]
```